core/embed/sec/tropic/tropic_transport.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <sec/tropic_transport.h>
+#include <sec/secret.h>
+#include "ed25519-donna/ed25519.h"
+
+#define PKEY_INDEX_BYTE PAIRING_KEY_SLOT_INDEX_0
+
+STATIC lt_handle_t lt_handle = {0};
+
+tropic_result tropic_init(void) {
+  lt_ret_t ret = lt_init(&lt_handle);
+  if (ret != LT_OK) {
+    return TROPIC_ERR_INIT;
+  }
+
+  return TROPIC_SUCCESS;
+}
+
+tropic_result tropic_handshake(const uint8_t *trezor_privkey) {
+  lt_ret_t ret = LT_FAIL;
+
+  uint8_t X509_cert[LT_L2_GET_INFO_REQ_CERT_SIZE] = {0};
+  ret = lt_get_info_cert(&lt_handle, X509_cert, LT_L2_GET_INFO_REQ_CERT_SIZE);
+  if (ret != LT_OK) {
+    return TROPIC_ERR_GET_INFO_CERT;
+  }
+
+  uint8_t stpub[32] = {0};
+  ret = lt_cert_verify_and_parse(X509_cert, 512, stpub);
```

Comment:
Does the Tropic emulator use a constant public key in the certificate?

Reply:
TROPIC01 model typically accepts config files during start, they consist of certificate* and STPRIV. Technically speaking those might be regenerated, but current model have them commited in repo, therefore "static". So I would say, your model is using probably still those "hardcoded" commited files.

*this will slightly change, now there is just chip's certificate but we will provide chain of certificates.

Reply:
Something like this, I assume? https://github.com/trezor/trezor-firmware/pull/4172/commits/18e8049546088c05ef0c8d3dc67bebb8a51cf4a6

Basically I ran `lt_session_start` & `lt_get_info_cert` once, dumped the key, then added it as a constant which we can now use. So essentially the pairing was done once - and it will be done once for every individual Tropic chip.

PS: I also removed the Tropic related code from `kernel/main.c` for now - since it was not being built and there is no way to test it anyway! We should finalize it for the emulator first, then we can port the code to run on a physical device.